### PR TITLE
ADD USER is only visible to Admin

### PR DESCRIPTION
**Pull request by KiGrant**
I added changes to the actions.payload in the sharedReducer.js file. I also did some more rendering changes in the profile components. I'm having trouble figuring out how to render the edit button when a user is logged in, and see there profile while searching. Any ideas on how I can accomplish this?

### CODE REVIEW

**Comment by matinaspatsos** [comment]
Kishon,

You can use the _graduateId_ to control whether or not an edit button is visible. The _graduteId_ returned in the login AJAX request is the graduateId of the user's graduate profile. If a user is a student and they have a  _graduateId_, then they should only be able to edit the profile with the same id.

Alos, students will need to be able to add their own profiles, so the add profile button should be visible (if they do not have one already).

File: src/components/EditProfile/EditProfile.jsx [hunk 1]
```diff
@@ -64,9 +64,10 @@ class EditProfile extends Component {
 
   // Setting Password with regexp
   passwordValidation = () => {
+    const { password, confirmPassword } = this.state;
```
[the remark this answers is not in the document]
Comment (KiGrant) [reply src/components/EditProfile/EditProfile.jsx]: Thanks I'll try to fix that bug

**Comment by lindakovacs** [comment]
Tested locally and I was able to reset the password as a student from the edit profile.

File: src/components/NewProfile/NewProfile.jsx [hunk 2]
```diff
@@ -58,15 +68,27 @@ class NewProfile extends Component {
...
     e.preventDefault();
-
+    // Setting Password with regexp
+    // let regexp = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[^a-zA-Z0-9])(?!.*\s).{6,99}$/;
+    const { password, confirmPassword } = this.state.profileData;
+    if (password === confirmPassword) {
+      return true;
+    } else {
+      alert("doesn't match");
+    }
```
Comment (matinaspatsos) [inline src/components/NewProfile/NewProfile.jsx]: Sorry, reverting to old code so that the form will submit

File: src/sharedReducer.js [hunk 3]
```diff
@@ -5,19 +5,26 @@ const sharedReducer = (
     profiles: null,
     isAdmin: !!localStorage.token,
     isLoading: false,
-    hasError: false
+    hasError: false,
+    isGrad: !!localStorage.token,
+    graduateId: ""
   },
   action
 ) => {
```
Comment (matinaspatsos) [inline src/sharedReducer.js]: We need to store graduate ID here

**Pull request merged**